From a9be588abe9027c99962c914c280bc1531ab7dc7 Mon Sep 17 00:00:00 2001
From: Techcable <Techcable@outlook.com>
Date: Tue, 6 Jan 2015 22:12:31 -0600
Subject: [PATCH] POM changes

Check our github versioning
Add a dependency on jsr305 (@Nullable, @NonNull)

diff --git a/pom.xml b/pom.xml
index d8dc598..4dccd64 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4,18 +4,18 @@
     <modelVersion>4.0.0</modelVersion>
 
     <parent>
-        <groupId>com.destroystokyo.paper</groupId>
-        <artifactId>paper-parent</artifactId>
+        <groupId>net.techcable.tacospigot</groupId>
+        <artifactId>parent</artifactId>
         <version>dev-SNAPSHOT</version>
     </parent>
 
-    <groupId>com.destroystokyo.paper</groupId>
-    <artifactId>paper-api</artifactId>
-    <version>1.9.4-R0.1-SNAPSHOT</version>
+    <groupId>net.techcable.tacospigot</groupId>
+    <artifactId>tacospigot-api</artifactId>
+    <version>1.9.4-R0.1</version>
     <packaging>jar</packaging>
 
-    <name>Paper-API</name>
-    <url>https://github.com/PaperSpigot/Paper</url>
+    <name>TacoSpigot-API</name>
+    <url>https://github.com/TacoSpigot/TacoSpigot</url>
     <description>An enhanced plugin API for Minecraft servers.</description>
 
     <properties>
@@ -125,6 +125,12 @@
             <artifactId>asm-all</artifactId>
             <version>5.0.4</version>
         </dependency>
+        <!-- @NonNull, @Nullable -->
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+            <version>1.3.9</version>
+        </dependency>
     </dependencies>
 
     <build>
diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index 6d32e1d..32177d3 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -191,8 +191,10 @@ public class VersionCommand extends BukkitCommand {
     private void obtainVersion() {
         String version = Bukkit.getVersion();
         if (version == null) version = "Custom";
-        if (version.startsWith("git-Paper-")) {
-            String[] parts = version.substring("git-Paper-".length()).split("[-\\s]");
+        // TacoSpigot start
+        if (version.startsWith("git-TacoSpigot-")) {
+            String[] parts = version.substring("git-TacoSpigot-".length()).split("[-\\s]");
+            // TacoSpigot end
             int distance = getDistance(null, parts[0]);
             switch (distance) {
                 case -1:
@@ -205,7 +207,7 @@ public class VersionCommand extends BukkitCommand {
                     setVersionMessage("Unknown version");
                     break;
                 default:
-                    setVersionMessage("You are " + distance + " version(s) behind");
+                    setVersionMessage("You are " + distance + " commit(s) behind"); // TacoSpigot - version -> commit
             }
         } else if (version.startsWith("git-Bukkit-")) {
             // Paper end
@@ -243,13 +245,17 @@ public class VersionCommand extends BukkitCommand {
 
     // Paper start
     private static int getDistance(String repo, String verInfo) {
+        // TacoSpigot start - remove jenkins check
+        /*
         try {
             int currentVer = Integer.decode(verInfo);
             return getFromJenkins(currentVer);
         } catch (NumberFormatException ex) {
+            */
             verInfo = verInfo.replace("\"", "");
-            return getFromRepo("PaperMC/Paper", verInfo);
-        }
+            return getFromRepo("TacoSpigot/TacoSpigot", verInfo); // 'PaperMC/Paper' -> 'TacoSpigot/TacoSpigot'
+        //}
+        // TacoSpigot end
             /*
             BufferedReader reader = Resources.asCharSource(
                     new URL("https://hub.spigotmc.org/stash/rest/api/1.0/projects/SPIGOT/repos/" + repo + "/commits?since=" + URLEncoder.encode(hash, "UTF-8") + "&withCounts=true"),
@@ -267,6 +273,8 @@ public class VersionCommand extends BukkitCommand {
             */
     }
 
+    // TacoSpigot start - remove jenkins check
+    /*
     private static int getFromJenkins(int currentVer) {
         try {
             BufferedReader reader = Resources.asCharSource(
@@ -287,6 +295,8 @@ public class VersionCommand extends BukkitCommand {
             return -1;
         }
     }
+    */
+    // TacoSpigot end
 
     // Contributed by Techcable <Techcable@outlook.com> in GH PR #65
     private static final String BRANCH = "master";
@@ -306,7 +316,7 @@ public class VersionCommand extends BukkitCommand {
                     case "behind":
                         return ((Number) obj.get("behind_by")).intValue();
                     default:
-                        return -1;
+                        return -2;
                 }
             } catch (ParseException | NumberFormatException e) {
                 e.printStackTrace();
-- 
2.8.2.windows.1

